regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
+        "HEAT Value"            = "slvalue_heat",
+        "CO2 reduction"         = "sico2"
+      )
+
+    }
+
+    # Update line options available, and set to default if the currently-selected option does not exist
+    if (input$scenario %in% local_scenarios) {
+      updateSelectInput(session, "scenario", choices = local_scenarios, selected = input$scenario)
+    } else {
+      updateSelectInput(session, "scenario", choices = local_scenarios)
+    }
+
+    if (input$line_type %in% local_line_types) {
+      updateSelectInput(session, "line_type", choices = local_line_types, selected = input$line_type)
+    } else {
+      updateSelectInput(session, "line_type", choices = local_line_types)
+    }
+
+    if (input$line_order %in% local_line_order) {
+      updateSelectInput(session, "line_order", choices = local_line_order, selected = input$line_order)
+    } else {
+      updateSelectInput(session, "line_order", choices = local_line_order)
+    }
+
+  }
+
+  ##############
+  # Initialise region and to_plot, and update right hand menu by region/purpose
+  ##############
+
+  ## Create region, to_plot and (for persistent geographical values) helper
+  region <- reactiveValues(current = NA, data_dir = NA, geography = NA, repopulate_region = F, purposes_present = NA)
+  to_plot <- NULL
+  helper <- NULL
+  helper$e_lat_lng <- ""
+
+  ## Set  values of region
   observe({
-    query <- parseQueryString(session$clientData$url_search)
+    region$current
+    region$data_dir
+    region$geography
+    region$repopulate_region
+    region$purposes_present
+    input$purpose
 
+    # Identify region
+    query <- parseQueryString(session$clientData$url_search)
     if (is.na(region$current)) {
-      region$current <- if(isTRUE(query[['r']] %in% regions$Region)){
+      region$current <- if (isTRUE(query[['r']] %in% regions$region_name)) {
         query[['r']]
-      } else if (exists("starting_region")) {
-        starting_region
       } else {
-        "west-yorkshire"
+        "isle-of-wight"
       }
-      region$data_dir <- file.path(data_dir_root, region$current)
-      region$all_trips <- dir.exists(file.path(data_dir_root, region$current , 'all-trips'))
     }
 
+    # Notify browser to update URL to reflect new region
     session$sendCustomMessage("regionchange", region$current)
-    region$data_dir
-    region$repopulate_region
 
-    to_plot$straight_line <<- readRDS(file.path(region$data_dir, "l.Rds"))
-    to_plot$zones <<- readRDS(file.path(region$data_dir, "z.Rds"))
-    to_plot$cents <<- readRDS(file.path(region$data_dir, "c.Rds"))
+    # Define region geography, forcing a default in cases where the geography is not available
+    if (input$purpose =="commute") {
+      if (input$geography %in% c("msoa", "lsoa")) {
+        region$geography <<- input$geography
+      } else {
+        region$geography <<- "msoa"
+      }
+    } else if (input$purpose =="school") {
+      if (input$geography %in% c("lsoa")) {
```

Comment:
Is there any point in this `if` as it is always setting the `geography` to `"lsoa"`?

Reply:
Something like:

```R
purpose_geog <- data.frame(
  "commute" = c("msoa", "lsoa"),
  "school" = c("lsoa"),
  "alltrips" = c("msoa"),
)

if (input$geography %in% purpose_geog[input$purpose]  {
  region$geography <<-  input$geography
} else {
  purpose_geog[input$purpose][1]
}

Reply:
yes, that looks like it would be a much more elegant way to get to the same place